src/Plugin/Field/FieldWidget/FileEntityBrowser.php

```diff
@@ -0,0 +1,212 @@
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    $settings = parent::defaultSettings();
+    $settings['view_mode'] = 'thumbnail';
+    return $settings;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $element = parent::settingsForm($form, $form_state);
+
+    $element['field_widget_display']['#access'] = FALSE;
+    $element['field_widget_display_settings']['#access'] = FALSE;
+
+
+    $element['view_mode'] = [
+      '#title' => t('File view mode'),
+      '#type' => 'select',
+      '#default_value' => $this->getSetting('view_mode'),
+      '#options' => \Drupal::service('entity_display.repository')->getViewModeOptions('file'),
+    ];
+
+    return $element;
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() {
+    $summary = [];
+    $entity_browser_id = $this->getSetting('entity_browser');
+    $view_mode = $this->getSetting('view_mode');
+
+    if (empty($entity_browser_id)) {
+      return [t('No entity browser selected.')];
+    }
+    else {
+      $browser = $this->entityManager->getStorage('entity_browser')
+        ->load($entity_browser_id);
+      $summary[] = t('Entity browser: @browser', ['@browser' => $browser->label()]);
+    }
+
+    if (!empty($view_mode)) {
+      $summary[] = t('View mode: @name', ['@name' => $view_mode]);
+    }
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function displayCurrentSelection($details_id, $field_parents, $entities, FieldItemListInterface $items) {
+
+    $view_mode = $this->getSetting('view_mode');
+
+    $view_builder = \Drupal::entityTypeManager()->getViewBuilder('file');
+    $config = \Drupal::config('file.settings');
+
+    $delta = 0;
+
+    $order_class = $this->fieldDefinition->getName() . '-delta-order';
+
+    $current = [
+      '#type' => 'table',
+      '#header' => [$this->t('Preview'), $this->t('Metadata'), ['data' => $this->t('Operations'), 'colspan' => 2], t('Order', array(), array('context' => 'Sort order'))],
+      '#empty' => $this->t('No files yet'),
+      '#attributes' => ['class' => ['entities-list']],
+      '#tabledrag' => array(
+        array(
+          'action' => 'order',
+          'relationship' => 'sibling',
+          'group' => $order_class,
+        ),
+      ),
+    ];
+    foreach ($entities as $entity) {
+      $display = $view_builder->view($entity, $view_mode);
+
+      // Find the default description.
+      $description = '';
+      $weight = $delta;
+      foreach ($items as $item) {
+        if ($item->target_id == $entity->id()) {
+          $description = $item->description;
+          $weight = $item->_weight ?: $delta;
+        }
+      }
+
+      $current[$entity->id()] = [
+        '#attributes' => [
+          'class' => ['draggable'],
+          'data-entity-id' => $entity->id()
+        ],
+        'display' => $display,
+        'meta' => [
```

Comment:
Added that now, will push soon. Looks like this, with both enabled and required:

![selection_018](https://cloud.githubusercontent.com/assets/40826/12558769/b7f4dc12-c391-11e5-82e2-ae0ed1894819.png)

Also respecting the settings for showing/requiring those 3 fields.